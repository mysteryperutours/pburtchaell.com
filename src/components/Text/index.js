import React, {createElement, Element} from 'react'
import PropTypes from 'prop-types'
import Link from 'gatsby-link'
import * as textTypes from './types'

const Text = (props) => {
  const {
    type,
    style,
    children,
    className,
    onClick,
  } = props

  const elementProps = {
    style,
    className,
    onClick,
  }

  return createElement(
    type,
    elementProps,
    children,
  )
}

Text.propTypes = {
  type: PropTypes.oneOf([
    textTypes.HEADER_1,
    textTypes.HEADER_2,
    textTypes.HEADER_3,
    textTypes.BODY,
    textTypes.SMALL,
  ]).isRequired,
  style: PropTypes.object,
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.arrayOf(PropTypes.node),
  ]),
  onClick: PropTypes.func,
}

Text.defaultProps = {
  type: textTypes.BODY,
}

export {Text as default, textTypes as types}
