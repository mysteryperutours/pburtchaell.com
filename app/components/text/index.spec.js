import React from 'react';
import Text, { types } from './index';

describe(Text.name, () => {
  const DEFAULT_CHILD = 'foo';

  it('should render h1 element', () => {
    expect(shallow(
      <Text type={types.HEADER_1}>
        My title
      </Text>
    )).toMatchSnapshot();
  });

  it('should render h2 element', () => {
    expect(shallow(
      <Text type={types.HEADER_2}>
        {DEFAULT_CHILD}
      </Text>
    )).toMatchSnapshot();
  });

  it('should render h3 element', () => {
    expect(shallow(
      <Text type={types.HEADER_3}>
        {DEFAULT_CHILD}
      </Text>
    )).toMatchSnapshot();
  });

  it('should render paragraph element', () => {
    expect(shallow(
      <Text type={types.BODY}>
        {DEFAULT_CHILD}
      </Text>
    )).toMatchSnapshot();
  });
});
