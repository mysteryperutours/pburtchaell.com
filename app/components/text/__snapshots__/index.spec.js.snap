// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render h1 element 1`] = `
<h1>
  foo
</h1>
`;

exports[`Text should render h2 element 1`] = `
<h2>
  foo
</h2>
`;

exports[`Text should render h3 element 1`] = `
<h3>
  foo
</h3>
`;

exports[`Text should render h4 element 1`] = `
<h4>
  foo
</h4>
`;

exports[`Text should render h5 element 1`] = `
<h5>
  foo
</h5>
`;

exports[`Text should render paragraph element 1`] = `
<p>
  foo
</p>
`;
